src/libraries/System.Private.Xml/src/System/Xml/Schema/CompiledidEntityConstraint.cs

```diff
@@ -52,11 +52,11 @@ public CompiledIdentityConstraint(XmlSchemaIdentityConstraint constraint, XmlNam
             //public Asttree (string xPath, bool isField, XmlNamespaceManager nsmgr)
             try
             {
-                _selector = new Asttree(constraint.Selector.XPath, false, nsmgr);
+                _selector = new Asttree(constraint.Selector!.XPath!, false, nsmgr);
```

Comment:
It might actually trigger in the tests (not sure) - the constructor will throw if you pass null though as it will try to compile it where it has a null check which throws.